Demo/Navigation/TurboNavigationController.swift

```diff
@@ -0,0 +1,111 @@
+        // This is a simplified version of how you might build out the routing
+        // and navigation functions of your app. In a real app, these would be separate objects
+        
+        // Dismiss any modals when receiving a new navigation
+        if presentedViewController != nil {
+            dismiss(animated: true)
+        }
+        
+        // Special case of navigating home, issue a reload
+        if url.path == "/", !viewControllers.isEmpty {
+            popViewController(animated: false)
+            session.reload()
+            return
+        }
+        
+        // - Create view controller appropriate for url/properties
+        // - Navigate to that with the correct presentation
+        // - Initiate the visit with Turbo
+        let viewController = makeViewController(for: url, properties: properties)
+        navigate(to: viewController, action: options.action, properties: properties)
+        visit(viewController: viewController, with: options, modal: isModal(properties))
+    }
+}
+
+extension TurboNavigationController {
+    
+    private func isModal(_ properties: PathProperties) -> Bool {
+        // For simplicity, we're using string literals for various keys and values of the path configuration
+        // but most likely you'll want to define your own enums these properties
+        let presentation = properties["presentation"] as? String
+        return presentation == "modal"
+    }
+    
+    private func makeViewController(for url: URL, properties: PathProperties = [:]) -> UIViewController {
+        // There are many options for determining how to map urls to view controllers
+        // The demo uses the path configuration for determining which view controller and presentation
+        // to use, but that's completely optional. You can use whatever logic you prefer to determine
+        // how you navigate and route different URLs.
+        
+        if let viewController = properties["view-controller"] as? String {
+            switch viewController {
+            case "numbers":
+                return NumbersViewController()
+            case "numbersDetail":
+                let alertController = UIAlertController(title: "Number", message: "\(url.path)", preferredStyle: .alert)
+                alertController.addAction(.init(title: "OK", style: .default, handler: nil))
+                return alertController
+            default:
+                assertionFailure("Invalid view controller, defaulting to WebView")
+            }
+        }
+            
+        return ViewController(url: url)
+    }
+    
+    private func navigate(to viewController: UIViewController, action: VisitAction, properties: PathProperties = [:], animated: Bool = true) {
+        // We support three types of navigation in the app: advance, replace, and modal
+        
+        if isModal(properties) {
+            if viewController is UIAlertController {
```

Comment:
Modified so alerts are presented correctly. They do not need to be embedded in a navigation controller.